ios/Classes/SwiftAirshipPlugin.swift

```diff
@@ -164,94 +164,101 @@ public class SwiftAirshipPlugin: NSObject, FlutterPlugin, PreferenceCenterOpenDe
             result(true)
         }
     }
-
+    
     private func takeOff(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
-        
-        guard let args = call.arguments as? [String: String] else {
+        AirshipLogger.debug("\(String(describing:call.arguments))")
+        guard let configDict = call.arguments as? FlutterStandardTypedData else {
             result(false)
-            return
+            fatalError("Invalid arguments: \(String(describing: call.arguments))")
+        }
+        do{
+            PluginStore.protoConfig = try AirshipConfig.init(serializedData:configDict.data)
+            
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.016) {  AirshipAutopilot.attemptTakeOff()
```

Comment:
Why did we need a delay here?

Reply:
Actually was trying to test, something I noticed, cold start is not always registering the channel on the first run. Try run it from scratch and let me know whether it is consistent every time.